cmd/cli/namespace_add.go

```diff
@@ -142,3 +151,20 @@ func (a *namespaceAddCmd) run() error {
 
 	return nil
 }
+
+// meshExists determines if a mesh with meshName exists within the cluster
+func meshExists(clientSet kubernetes.Interface, meshName string) (bool, error) {
+	// search for the mesh across all namespaces
+	deploymentsClient := clientSet.AppsV1().Deployments("")
+	// search and match using the mesh name provided
+	labelSelector := metav1.LabelSelector{MatchLabels: map[string]string{"meshName": meshName}}
+	listOptions := metav1.ListOptions{
+		LabelSelector: labels.Set(labelSelector.MatchLabels).String(),
+	}
+	osmControllerDeployments, err := deploymentsClient.List(context.TODO(), listOptions)
+	if err != nil {
+		return false, errors.Errorf("Cannot obtain information about the mesh [%s]: [%v]", meshName, err)
+	}
+	// the mesh is present if there are osm controllers for the mesh
+	return len(osmControllerDeployments.Items) != 0, nil
+}
```

Comment:
@shashankram & @jaellio: As per both of your suggestions, I have simplified + clarified the code that determines whether the mesh exists within the cluster.

Reply:
For documentation: I suggested removing the namespace parameter from getMeshController since this value would only be "" to search all namespaces for an existing mesh.